File: programmers/Lv3/네트워크.cpp
#include <string>
#include <iostream>
#include <vector>

using namespace std;
vector<int>v[201];
bool visit[201]={false,};
void dfs(int cur){
    visit[cur] = true;
    
    for(int i=0;i<v[cur].size();i++){
        int next = v[cur][i];
        if(!visit[next]){
            dfs(next);
        }
    }
}
int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    for(int i=0;i<computers.size();i++){
        for(int j=0;j<computers[i].size();j++){
            if(!computers[i][j])continue;
            if(i == j)continue;
            if(computers[i][j] == 1){
                v[i].push_back(j);
                v[j].push_back(i);
            }
        }
    }
    for(int i=0;i<n;i++){
        if(!visit[i]){
            dfs(i);
            answer++;
        }
    }
    return answer;
}

File: programmers/Lv3/네트워크.js
function solution(n, computers) {
    let answer = 0;
    let visit = Array(n).fill(false);
    for(let i=0;i<n;i++){
        if(!visit[i]){
            dfs(i);
            answer += 1;
        }
    }
    
    function dfs(cur){
        for(let i=0;i<computers[cur].length;i++){
            let next = computers[cur][i];
            if(!visit[i] && next === 1){
                visit[i] = true;
                dfs(i);
            }
        }
    }
    return answer;
}
